.github/actions/publish-image/action.yml

```diff
@@ -59,62 +59,124 @@ runs:
         echo "IMAGE_NAME=$IMAGE_NAME" >> $GITHUB_ENV
         echo "TAG_NAME=$TAG_NAME" >> $GITHUB_ENV
 
+    # TODO (@NickLarsenNZ): Make this a reusable action in .github/actions/push-and-sign/action.yml
     - name: Push Image to repo.stackable.tech and sign via cosign
       shell: bash
       run: |
         set -euo pipefail
-        docker image push "$(< bake-target-tags)"
+        TARGET_TAG="$(< bake-target-tags)"
+        docker image push "$TARGET_TAG"
         # Obtain the digest of the image, because signing by tag is deprecated and will be removed from cosign in the future
-        DIGEST=$(docker images --digests "$(< bake-target-tags)" --format '{{.Digest}}')
-        echo "DIGEST=$DIGEST" >> $GITHUB_ENV
+        # This must be done after a push, otherwise the repo digest is empty.
+        # Find the digest for the image. Notice the architecture at the end of TAG
+        # IMAGE_VERSION does not contain the architecture (otherwise we could have used ${IMAGE_NAME}:${IMAGE_VERSION})
+        # Given:
+        # IMAGE_NAME: docker.stackable.tech/stackable/hello-world
+        # TARGET_TAG: docker.stackable.tech/stackable/hello-world:0.0.1-SNAPSHOT-stackable0.0.0-dev-arm64
+        # Expect:
+        # STDOUT:     docker.stackable.tech/stackable/hello-world@sha256:917f800259ef4915f976e93987b752fd64debf347568610d7f685d20220fc88a
+        REPO_DIGEST=$(
+          docker inspect "$TARGET_TAG" --format json | \
+          jq -r \
+          --arg IMAGE_NAME "$IMAGE_NAME" \
+          --arg TARGET_TAG "$TARGET_TAG" \
+          '
+              map(select(.RepoTags[] | contains($TARGET_TAG)))[0]
+              | .RepoDigests[]
+              | select(. | startswith($IMAGE_NAME))
+          '
+        )
```

Comment:
I think `docker inspect "$TARGET_TAG" --format '{{index .RepoDigests 0}}'` would yield the same result in our case. But I'm fine with that variant as well since it's more strict, even though it's also more complex.

Reply:
Yeah, I just wanted the extra safety of getting the digest that matches, rather than using the first element.